src/odemis/gui/comp/overlay/world.py

```diff
@@ -833,48 +835,61 @@ def deactivate(self):
         WorldOverlay.deactivate(self)
 
 
-NONE_RULER_MODE = 0
-MOVE_RULER_MODE = 1
-EDIT_START_RULER_MODE = 2
-EDIT_END_RULER_MODE = 3
-
-class Ruler(object):
+class GadgetToolInterface(with_metaclass(ABCMeta, object)):
     """
-    Represent a "ruler" in the canvas (as a sub-part of the RulerOverlay). Used to draw the ruler
-    and also handle the mouse interaction when dragging/moving the ruler.
+    This abstract GadgetToolInterface class forms the base for a series of classes that
+    refer to gadgets tools and their functionality.
     """
 
+    def __init__(self, cnvs):
+        """ Args: (cnvs) canvas passed by the GadgetOverlay and used to draw the gadgets """
+        self.cnvs = cnvs
+
+    def start_dragging(self, drag, vpos, tool_mode):
+        """ The user can start dragging the tool when the left mouse button is pressed down """
```

Comment:
Please put the full docstring, with the argument and return descriptions here. In the subclasses, you can keep the docstring empty, or much shorter.